include/hve/arch/intel_x64/irq.h

```diff
@@ -0,0 +1,159 @@
+//
+// Bareflank Extended APIs
+//
+// Copyright (C) 2018 Assured Information Security, Inc.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#ifndef IRQ_HDLR_INTEL_X64_EAPIS_H
+#define IRQ_HDLR_INTEL_X64_EAPIS_H
+
+#include <list>
+#include <array>
+#include <utility>
+
+#include <bfgsl.h>
+#include <bfdebug.h>
+#include <bfvmm/hve/arch/intel_x64/vmcs/vmcs.h>
+#include <bfvmm/hve/arch/intel_x64/exit_handler/exit_handler.h>
+
+// -----------------------------------------------------------------------------
+// Exports
+// -----------------------------------------------------------------------------
+
+#include <bfexports.h>
+
+#ifndef STATIC_EAPIS_HVE
+#ifdef SHARED_EAPIS_HVE
+#define EXPORT_EAPIS_HVE EXPORT_SYM
+#else
+#define EXPORT_EAPIS_HVE IMPORT_SYM
+#endif
+#else
+#define EXPORT_EAPIS_HVE
+#endif
+
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4251)
+#endif
+
+namespace eapis
+{
+namespace intel_x64
+{
+
+namespace reason = ::intel_x64::vmcs::exit_reason::basic_exit_reason;
+
+// -----------------------------------------------------------------------------
+// Definitions
+// -----------------------------------------------------------------------------
+
+///
+/// External interrupt handler
+///
+class EXPORT_EAPIS_HVE irq
```

Comment:
this should be renamed to "external_interrupt", and then all references to irq should be changed to that. for this class. I would prefer that we stick to the naming the Intel uses so that we do not get lost. New users are going to be reading the manual, so we want them to see the manual, and then see the code an know what they are looking at.

Reply:
I assume the crs class will be renamed to control_register_access as well?